jsonifier/models/models.py

```diff
@@ -194,6 +201,62 @@ def jsonify(self, parser, one=False):
              'field_name:json_key'
         ]
 
+        if with_fieldname=True,the name of the field will be added
+        to the value. the name of the field will be translated.
+
+        rec.jsonify(
+            parser=['name', 'create_date',
+                    'order_line', ['id' , 'product_uom', 'is_expense'])],
+            with_fieldname=False)
+
+        returns:
+
+        [{'name': 'SO/2022/006',
+                   'create_date': '2022-10-14T15:55:39.823019+02:00',
+                   'order_line': {
+                         'id': 4,
+                         'product_uom': 'Stuk(s)
+                         'is_expense': False}
+                    }]
+
+        rec.jsonify(
+            parser=['name', 'create_date',
+                    'order_line', ['id' , 'product_uom', 'is_expense'])],
+            with_fieldname=True)
+
+        would give us:
+        [{'name': { value: 'SO/2022/006', 'fieldname': 'Name'},
+          'create_date': {
+                         'value': '2022-10-14T15:55:39.823019+02:00',
+                         'fieldname': 'Created On'},
+          'order_line': {'value': {
+                             'id': {'value':  4, 'fieldname': 'ID'},
+                             'Product_uom': {'value': 'Stuks',
+                                             'fieldname': 'Unit Of measure'},
+                             'is_expense': {'value': False  ,
+                                            'fieldname': 'Is an Expense'}}
+                         },
+                         fieldname: 'Sale Order Lines'
+              }
+        ]
+
+        therefore the value of the field will be accessed not anymore as:
+
+                result['name']
+                result['order_line']['product_uom']
+
+        but:
+
+                result['name']['value']  # get value of field.name
+                result['name']['fieldname']   # get the translated fieldname of name
+                # get value of field.order_line.product_uom.
+                result['order_line']['value']['product_uom']['value']
+                # fieldname of  field.order_line.product_uom
+                result['order_line']['value']['product_uom']['fieldname']
+
+        I need to use structure {'value':x, 'fieldname': y} because the json name
+        (order_line) may be duplicated in the various levels of relational fields.
```

Comment:
We went for the nested dictionary approach, but maybe it would have been better to go for sister keys at the same level named `_fieldname_$field` instead; so that the with_fieldname context key won't break code that isn't expecting the context key to be there.

Reply:
$field_fieldname is now  a "sister key".  but now i see that fieldname_$field is better. Changing.